cmd/snap/main_test.go

```diff
@@ -62,3 +65,26 @@ func (s *SnapSuite) Stdout() string {
 func (s *SnapSuite) Stderr() string {
 	return s.stderr.String()
 }
+
+func (s *SnapSuite) RedirectClientToTestServer(handler func(http.ResponseWriter, *http.Request)) {
+	server := httptest.NewServer(http.HandlerFunc(handler))
+	s.BaseTest.AddCleanup(func() { server.Close() })
+	ClientConfig.BaseURL = server.URL
+	s.BaseTest.AddCleanup(func() { ClientConfig.BaseURL = "" })
+}
+
+// DecodedRequestBody returns the JSON-decoded body of the request.
+func DecodedRequestBody(r *http.Request, c *C) map[string]interface{} {
+	var body map[string]interface{}
+	decoder := json.NewDecoder(r.Body)
+	err := decoder.Decode(&body)
+	c.Assert(err, IsNil)
+	return body
+}
+
+// EncodeResponseBody writes JSON-serialized body to the response writer.
+func EncodeResponseBody(w http.ResponseWriter, c *C, body interface{}) {
```

Comment:
c *C as 2nd argument is strange, any particular reasons?

seems more common to be uniform and put first or last